src/NzbDrone.Core/Configuration/ConfigFileProvider.cs

```diff
@@ -391,10 +392,25 @@ private string GenerateApiKey()
             return Guid.NewGuid().ToString().Replace("-", "");
         }
 
+        private void ValidateApiKey()
+        {
+            var apiKey = ApiKey;
+
+            if (new Regex("^[a-zA-Z0-9]{20,}$").IsMatch(apiKey))
```

Comment:
Do we want to enforce 20? This may kill alot of instances for people that are using custom keys (we know they were many based on some previous issues). 

Could we do something like at least 8, then health warning from 8-20? @markus101 thoughts here?

Reply:
For Sonarr at least I think it’s fine since it’s going to hit in a major release, but is a somewhat breaking change so I can see showing a health check warning instead.

Reply:
True, I forget we are not released on 4 yet. We'll see if we here anything, in which case we can advise they can still set it during deployment, just needs to be 20+

Reply:
30 might be a bit high for some for a health nag? I'm fine with it, just anticipating the issues.

Also Im not against the regenerate for very poor keys, just need to know that we might get some backlash from people setting via auto deployment.

Reply:
Yea, the check for now seems sane